servers/src/main/java/tachyon/worker/block/BlockServiceHandler.java

```diff
@@ -195,9 +195,8 @@ public boolean promoteBlock(long blockId) throws TachyonTException, ThriftIOExce
   public String requestBlockLocation(long sessionId, long blockId, long initialBytes)
       throws TachyonTException, ThriftIOException {
     try {
-      // NOTE: right now, we ask allocator to allocate new blocks in MEM tier
-      return mWorker.createBlock(sessionId, blockId, StorageLevelAlias.MEM.getValue(),
-          initialBytes);
+      // NOTE: right now, we ask allocator to allocate new blocks in any tier
+      return mWorker.createBlock(sessionId, blockId, -1 /* any tier */, initialBytes);
```

Comment:
Or is there an implicit assumption that the tiered store will first fill the the top tier?